sbnanalysis/ana/SBNOsc/NumuSelection.cxx

```diff
@@ -37,31 +145,228 @@ bool NumuSelection::ProcessEvent(const gallery::Event& ev, std::vector<Event::In
+      visible_E += this_visible_energy;
+      smeared_visible_E += this_smeared_visible_energy; 
+    }
+  }
+  // now the showers
+  for (auto const &mcs: mcshower_list) {
+    if (isFromNuVertex(mctruth, mcs)) {
+      double mass = PDGMass(mcs.PdgCode());
+      double this_visible_energy = mcs.Start().E() - mass;
+      double this_smeared_visible_energy = this_visible_energy + _smear->Smear(this_visible_energy, mcs.PdgCode(), contained_in_FV);
+      visible_E += this_visible_energy;
+      smeared_visible_E += this_smeared_visible_energy; 
+    }
+  }
+
+
+  // convert visible energies to GeV
+  return {contained_in_FV, l_contained_length, l_length, visible_E / 1000., smeared_visible_E / 1000., smeared_eccqe}; 
+}
+
+std::vector<bool> NumuSelection::Select(const gallery::Event& ev, const simb::MCTruth& mctruth, unsigned truth_ind, const NumuSelection::NuMuInteraction &intInfo) {
+  // get the neutrino
+  const simb::MCNeutrino& nu = mctruth.GetNeutrino();
 
-  bool selected = !reco.empty();
+  // select true CC events
+  bool pass_true_CC = (nu.CCNC() == simb::kCC && (nu.Mode() == 0 || nu.Mode() == 10) && nu.Nu().PdgCode() == 14);
 
-  if (selected) {
-    fNuCount++;
+  // pass fiducial volume cut
+  bool pass_FV = passFV(nu.Nu().Position().Vect());
+
+  // min length cut
+  bool pass_min_length = passMinLength(intInfo.l_contained_length, intInfo.l_is_contained);
+
+  // pass vertex reconstruction cut
+  bool pass_reco_vertex = true;
+  if (_config.vertexDistanceCut > 0) {
+    double truth_v[3];
+    truth_v[0] = nu.Nu().Vx();
+    truth_v[1] = nu.Nu().Vy();
+    truth_v[2] = nu.Nu().Vz();
+  
+    // get the reco vertex information
+    auto const pfp_handle = ev.getValidHandle<std::vector<recob::PFParticle>>("pandoraNu");
+    art::FindMany<recob::Vertex> fvtx(pfp_handle, ev, "pandoraNu");
+    // index of truth info is same as index of vertex info
+    std::vector<const recob::Vertex*> vertices = fvtx.at(truth_ind);
+    // neutrino will only have one associated vetex
+    auto vertex = vertices[0]->position();
+    TVector3 reco_v(vertex.X(), vertex.Y(), vertex.Z());
+    
+    pass_reco_vertex = passRecoVertex(nu.Nu().Position().Vect(), reco_v);
   }
 
-  return selected;
+  // print selection information
+  if (_config.verbose) {
+    std::cout << "NEW EVENT" << std::endl;
+    std::cout << "CCNC: " << nu.CCNC() << " MODE: " << nu.Mode() << " PDG: " << nu.Nu().PdgCode() << std::endl;
+    std::cout << "pass CC: " << pass_true_CC << std::endl;
+    std::cout << "Pos: " << nu.Nu().Vx() << " " << nu.Nu().Vy() << " " << nu.Nu().Vz() << std::endl;
+    std::cout << "pass FV: " << pass_FV << std::endl;
+    std::cout << "pass Reco: " << pass_reco_vertex << std::endl;
+  }
+
+  // retrun list of cuts
+  return {pass_true_CC, pass_true_CC && pass_FV, pass_true_CC && pass_FV && pass_min_length, pass_true_CC && pass_FV && pass_reco_vertex};
```

Comment:
It would be nice to clean this up a bit, so that the return value is a bit more generic and makes the association between cuts and pass/fail manifest in a data structure. It doesn't look like the individual combinations actually get used anywhere, just an OR of them.

Reply:
The individual combinations get used for filling up different histograms -- I switched the vector to an array to make the nCuts thing more clear.

What do you mean by more generic?